src/main/java/org/jenkinsci/plugins/workflow/job/console/NewNodeConsoleNote.java

```diff
@@ -60,7 +60,7 @@
  * @see LogStorage#startStep
  */
 @Restricted(NoExternalUse.class)
-public class NewNodeConsoleNote extends ConsoleNote<WorkflowRun> {
+public class NewNodeConsoleNote extends ConsoleNote {
```

Comment:
This issue is kind of interesting. I am not sure if it has been reported previously, but any console notes that rely on a `Run` context totally break the per-step log view in the Pipeline steps view. (The notes aren't just skipped, they cause a `ClassCastException` [here](https://github.com/jenkinsci/jenkins/blob/7fb5f24efd92eda2461e9bf2f1fdc50295e12573/core/src/main/java/hudson/console/ConsoleAnnotationOutputStream.java#L119) and break all log display!)

Perhaps [`LogStorage.stepLog`](https://github.com/jenkinsci/workflow-api-plugin/blob/1679fa2f0f761c285a7309b8781c72e8fd9601c8/src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java#L115) should really return a `AnnotatedLargeText<WorkflowRun>` instead? Not sure if that can be changed compatibly at this point. `ConsoleAnnotationOutputStream` at least could probably be made more robust though.

There do not seem to be many implementations that would trigger the issue, see [here](https://github.com/search?type=code&q=+owner%3Ajenkinsci+consolenote%3CWorkflowRun) and [here](https://github.com/search?type=code&q=+owner%3Ajenkinsci+ConsoleNote%3CRun).

Reply:
Hmm, does this have anything to do with https://github.com/jenkinsci/jenkins/pull/3662?